Lib/test/test_asyncore.py

```diff
@@ -800,6 +801,36 @@ def test_quick_connect(self):
             if t.is_alive():
                 self.fail("join() timed out")
 
+    def test_map_altered_in_loop(self):
+        family = self.family
+        fail = self.fail
+
+        class NoWriteClient(BaseClient):
```

Comment:
Not sure that people will understand why this client is not writable. Maybe call this BateClient or PoisonedClient? 

And add docstring explaining why we need this.